### PR TITLE
Int b 18910 gun safe frontend

### PR DESCRIPTION
**Pull request by brooklyn-welsh**
## Other INT PR
#12458

## Summary

This branch adds onto the other INT PR to make the gun safe entitlement visible to the Prime when they are viewing a move's information.

### How to test

1. Access the Move Details page of any move as an SC
2. Scroll down to the allowances card, verify that there is a new row at the bottom for gun safe, and that it is "Unauthorized".
3. Click the link, check the box for "Gun safe authorized", then click submit.
4. Scroll back down to the "Allowances" card and verify that the Gun Safe value is now "Authorized".
5. Repeat steps 1-4 while logged in as a TOO.
6. Now the move should be available to Prime, login as Prime and view the move, near the end of the move info, under "Diversion", there should now be a new field titled "Gun Safe", which should display the value you entered as SC/TOO.

### CODE REVIEW

File: src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx [hunk 1]
```diff
@@ -23,7 +23,6 @@ import { setFlashMessage as setFlashMessageAction } from 'store/flash/actions';
 import scrollToTop from 'shared/scrollToTop';
 import { SIT_SERVICE_ITEMS_ALLOWED_UPDATE } from 'constants/serviceItems';
 import { MoveOrderDocumentType } from 'shared/constants';
-import { CHECK_SPECIAL_ORDERS_TYPES, SPECIAL_ORDERS_TYPES } from 'constants/orders';
```
Comment (taeJungCaci) [inline src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx:26]: Double checking if you meant to delete these - maybe from merge conflict? I feel like I saw these in merge conflicts.

File: src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx [hunk 2]
```diff
@@ -171,11 +170,9 @@ const MoveDetails = ({ setFlashMessage }) => {
...
           <div className="grid-col-12">
             <FlashGridContainer className={styles.flashContainer} data-testid="move-details-flash-grid-container">
-              {CHECK_SPECIAL_ORDERS_TYPES(moveTaskOrder?.order?.ordersType) ? (
-                <div className={styles.specialMovesLabel}>
-                  {SPECIAL_ORDERS_TYPES[`${moveTaskOrder?.order?.ordersType}`]}
-                </div>
-              ) : null}
+              {moveTaskOrder?.order?.ordersType !== 'BLUEBARK' ? null : (
+                <div className={styles.specialMovesLabel}>BLUEBARK</div>
+              )}
```
Comment (taeJungCaci) [inline src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx:175]: Same with these: Double checking if you meant to delete these - maybe from merge conflict? I feel like I saw these in merge conflicts.

File: src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx (hunk 1 above)
Comment (brooklyn-welsh) [reply src/pages/PrimeUI/MoveTaskOrder/MoveDetails.jsx:26]: Good catch, not sure how that got there, reverted a small change in the Main branch which I tried to carry over to INT, but apparently had this side-effect, just pushed a fix.

**Review by taeJungCaci — APPROVED** [review]
LGTM!

**Pull request merged**